x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -107,16 +107,18 @@ public JwtRealm(final RealmConfig realmConfig, final SSLService sslService, fina
             this.httpClient = null; // no setting means no HTTP client
         }
 
-        this.jwksAlgsHmac = this.parseJwksAlgsHmac(); // not reloadable
-        this.jwksAlgsPkc = this.parseJwksAlgsPkc(false); // reloadable
+        this.parseJwksAlgsHmac();
+        this.parseJwksAlgsPkc();
+        this.verifyAnyAvailableJwkAndAlgPair();
```

Comment:
Logging bug fix. If HMAC key filtered out (ex: too small), diagnostic log in PKC JWKSet method was being skipped because it was empty. I moved the checks to a separate method to ensure it gets printed. I found this issue while trying to generate the static realm config for the smoke test.